components/Search.js

```diff
@@ -23,25 +23,29 @@ function fetchData () {
 
 function Search () {
   // eslint-disable-next-line no-unused-vars
-  const [destination, setDestination] = React.useState("");
   const [buildingName, setBuildingName] = React.useState("");
+  const [from, setFrom] = React.useState("");
+  const [to, setTo] = React.useState("");
   const [data, setData] = React.useState();
 
-  let destinationName = destination.name;
+  let toName = to.name;
 
-  if (destinationName === undefined) {
-    destinationName = "";
-    AsyncStorage.setItem("destination", destinationName.toString());
+  if (toName === undefined) {
+    toName = "";
+    AsyncStorage.setItem("toLocation", toName.toString());
```

Comment:
We are starting to get too many AsyncStorage calls. Maybe we should consider implementing redux in our code.

Reply:
I agree. Either redux or upgrade Async. This needs to be an issue created.